[google_maps_flutter_platform_interface] Fix bitmap descriptor getter

MisTraleuh opened the pull request:
When you create a marker with a predefined asset like this, for example:

```dart
  Future<BitmapDescriptor> _getCustomMarker(bool isInPolygon, String assetFilePathColor, String assetFilePathGray) async {
    String imageAssetPath = isInPolygon ? assetFilePathColor : assetFilePathGray;
    
    return BitmapDescriptor.asset(
      const ImageConfiguration(size: Size(48, 48)),
      imageAssetPath,
    );
  }
```

Once the object is in our possession, we can't find the image assigned to this BitmapDescriptor. So I've added the ability to make a getter `getAssetName` to a ``BitmapDescriptor`` to retrieve the image in which it was built.

## Pre-Review Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or I have commented below to indicate which [version change exemption] this PR falls under[^1].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or I have commented below to indicate which [CHANGELOG exemption] this PR falls under[^1].
- [x] I updated/added any relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or I have commented below to indicate which [test exemption] this PR falls under[^1].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[^1]: Regular contributors who have demonstrated familiarity with the repository guidelines only need to comment if the PR is not auto-exempted by repo tooling.

<!-- Links -->
[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[version change exemption]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version
[following repository CHANGELOG style]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style
[CHANGELOG exemption]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog
[test exemption]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests

MisTraleuh commented:
Maybe I'm good no ?
![image](https://github.com/user-attachments/assets/5e776188-36cc-4962-9efd-24cec5b60935)

stuartmorgan-g commented:
Thanks for the contribution! You’ve checked boxes in the PR checklist above that are not reflected in this PR, so I’m assuming this is a work in progress and am marking it as a Draft. Please review the checklist, updating the PR as appropriate, and when the checklist reflects the state of the PR as posted please feel free to mark it as ready for review.

MisTraleuh commented:
> Thanks for the contribution! You’ve checked boxes in the PR checklist above that are not reflected in this PR, so I’m assuming this is a work in progress and am marking it as a Draft. Please review the checklist, updating the PR as appropriate, and when the checklist reflects the state of the PR as posted please feel free to mark it as ready for review.

Hello,
Thanks for your feedback, I didn't really understand. I checked off everything, didn't I? 
Sorry if I don't understand

Sincerely
Nathan

stuartmorgan-g commented:
> I didn't really understand. I checked off everything, didn't I?

Checking boxes without doing the corresponding step is not the goal of the checklist.

> _Replace this paragraph with a description of what this PR is changing or adding, and why. Consider including before/after screenshots._

This has not been completed.

> * [x]  I [linked to at least one issue that this PR fixes](https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview) in the description above.

This has not been completed.

Without either of these steps being completed, it is not clear what the goal of this PR is.

> * [ ]  I added new tests to check the change I am making, or I have commented below to indicate which [test exemption](https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests) this PR falls under[1](#user-content-fn-1-578b96550e8a9bed1578965c65f4c54e).

This has not been completed (and was checked when I left my comment).

MisTraleuh commented:
> > I didn't really understand. I checked off everything, didn't I?
> 
> Checking boxes without doing the corresponding step is not the goal of the checklist.
> 
> > _Replace this paragraph with a description of what this PR is changing or adding, and why. Consider including before/after screenshots._
> 
> This has not been completed.
> 
> > * [x]   I [linked to at least one issue that this PR fixes](https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview) in the description above.
> 
> This has not been completed.
> 
> Without either of these steps being completed, it is not clear what the goal of this PR is.
> 
> > * [ ]   I added new tests to check the change I am making, or I have commented below to indicate which [test exemption](https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests) this PR falls under[1](#user-content-fn-1-578b96550e8a9bed1578965c65f4c54e).
> 
> This has not been completed (and was checked when I left my comment).

Okay, I didn't really understand, I just added everything normally. Thanks a lot!

Thank you very much!
Nathan

MisTraleuh commented:
> > I didn't really understand. I checked off everything, didn't I?
> 
> Checking boxes without doing the corresponding step is not the goal of the checklist.
> 
> > _Replace this paragraph with a description of what this PR is changing or adding, and why. Consider including before/after screenshots._
> 
> This has not been completed.
> 
> > * [x]   I [linked to at least one issue that this PR fixes](https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview) in the description above.
> 
> This has not been completed.
> 
> Without either of these steps being completed, it is not clear what the goal of this PR is.
> 
> > * [ ]   I added new tests to check the change I am making, or I have commented below to indicate which [test exemption](https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests) this PR falls under[1](#user-content-fn-1-578b96550e8a9bed1578965c65f4c54e).
> 
> This has not been completed (and was checked when I left my comment).

Yeah now this is good normally ^^

stuartmorgan-g commented:
> * [x]  I [linked to at least one issue that this PR fixes](https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview) in the description above.

This has still not been done.

The PR description asserts that there's currently no way to get the information you want, but it's still not clear why (type switching seems like a straightforward solution), or what the use case you are trying to support is. One of the reasons we require an issue is so that we can discuss what the core issue is before jumping to a specific proposed solution.

Please file an issue describing the problem in detail, including the use case and why you cannot use type information to solve your use case without new APIs.

Pending that, closing this PR, as adding API to a base class that only applies to specific subclasses is generally an anti-pattern, so this is not something we would add without a compelling reason.
